fecgov/openFEC
# Fix "most recent" logic for Forms 1, 2, and 1M

## Summary (required)

- Resolves #3096 

**There should be only one `most_recent` F1 per committee and one F2 for candidates**

- Create a view for every materialized view that depends on `ofec_filings_amendments_all_vw`
- Modify amendment_chain logic for Forms 1, 1M, and 2
- Ensure single current version for F1, F1M and F2
- Add integration test for filing amendments, tests for amended F1s and F3s

committee filings page: '/committee/<committee_id>/filings/' > `filings.FilingsView` > `BaseFilings`
candidate filings page: '/filings/' > `filings.FilingsList` > `BaseFilings`

`BaseFilings` >  `ofec_filings_all_mv` > `ofec_filings_amendments_all_mv` > `disclosure.f_rpt_or_form_sub`

## How to test the changes locally

- Drop/Create `cfdm_test`/`invoke create_sample_db`
- `unset SQLA_CONN` and run your local api
- `unset FEC_API_URL` and run your local cms

Testing links: 
API
http://localhost:5000/v1/filings/?per_page=100
http://localhost:5000/v1/filings/?most_recent=true&per_page=100&form_type=F1
http://localhost:5000/v1/committee/C00559492/filings/?per_page=100
http://localhost:5000/v1/filings/?per_page=100&form_type=F2 (note that there isn't much F2 test data)
CMS
http://localhost:8000/data/committee/C00559492/?tab=filings&cycle=2016
http://localhost:8000/data/committee/C00559492/?tab=filings&cycle=2014

## Impacted areas of the application
List general components of the application that this PR will affect:

- Candidate filings tab
- Committee filings tab
- Processed filings in data tables
